ios/UARCTModule/UARCTMessageView.m

```diff
@@ -0,0 +1,175 @@
+/* Copyright Airship and Contributors */
+
+#import "UARCTMessageView.h"
+
+@interface UARCTMessageView()
+@property (nonatomic, strong) UAWKWebViewNativeBridge *nativeBridge;
+@property (nonatomic, strong) WKWebView *webView;
+@property (nonatomic, strong) UAInboxMessage *message;
+@property (nonatomic, strong) UADisposable *fetchMessagesDisposable;
+@end
+
+NSString *const UARCTMessageViewErrorMessageNotAvailable = @"MESSAGE_NOT_AVAILABLE";
+NSString *const UARCTMessageViewErrorFailedToFetchMessage = @"FAILED_TO_FETCH_MESSAGE";
+NSString *const UARCTMessageViewErrorMessageLoadFailed = @"MESSAGE_LOAD_FAILED";
+
+NSString *const UARCTMessageViewMessageKey = @"message";
+NSString *const UARCTMessageViewRetryableKey = @"retryable";
+NSString *const UARCTMessageViewErrorKey = @"error";
+
+@implementation UARCTMessageView
+
+- (instancetype) init {
+    self = [super initWithFrame:CGRectZero];
+    if (self) {
+        self.nativeBridge = [[UAWKWebViewNativeBridge alloc] init];
+        self.nativeBridge.forwardDelegate = self;
+        self.webView = [[WKWebView alloc] initWithFrame:self.bounds];
+        self.webView.navigationDelegate = self.nativeBridge;
+        self.webView.allowsLinkPreview = ![UAirship messageCenter].disableMessageLinkPreviewAndCallouts;
+        self.webView.configuration.dataDetectorTypes = WKDataDetectorTypeAll;
+        [self addSubview:self.webView];
+    }
+    return self;
+}
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    self.webView.frame = self.bounds;
+}
+
+- (void)setMessageID:(NSString *)messageID {
```

Comment:
where does this get called from?

Reply:
also, is there a need for the property? could you just pass it to loadMessage and mark it non-null?

Reply:
The message ID is a property on the js view object

```
<UAMessage
   messageId={state.messageId}
/>
```

To make that work, i had to use the macro to map the view property to the view:
```
RCT_REMAP_VIEW_PROPERTY(messageId, messageID, NSString)
```

I don't know if that accepts a method instead of a property, but since its a property in react it might as well be a property in obj-c